public/script.js

```diff
@@ -4607,7 +4607,9 @@ export function removeMacros(str) {
  * @returns {Promise<void>} A promise that resolves when the message is inserted.
  */
 export async function sendMessageAsUser(messageText, messageBias, insertAt = null, compact = false, name = name1, avatar = user_avatar) {
-    messageText = getRegexedString(messageText, regex_placement.USER_INPUT);
+    const insertAtDepth = (typeof insertAt === 'number' && insertAt >= 0 && insertAt <= chat.length) ? insertAt : null;
+
+    messageText = getRegexedString(messageText, regex_placement.USER_INPUT, { depth: insertAtDepth ?? 0 });
```

Comment:
It's not actually depth, it's an array index, so it should be reversed (chat.length = 0, etc).

Reply:
God, I even thought about that and then simply forgot. I really needed that nap.

Fixed now. I hope.

Reply:
Looks fine now.